### PR TITLE
Add JobRequests to the staff area

### PR DESCRIPTION
**Pull request by ghickman**
We can't lean on our typical filtering UI for JobRequests since we'd had to list nearly every User, Project, and Workspace.  I suspect we can add those in later with searchable dropdowns, however I didn't want this change to be bogged down by working out how to get those working in the staff area.

Fix: #3322

### CODE REVIEW

File: staff/views/job_requests.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            ]
+
+            # build up Q objects OR'd together.  We need to build them with
+            # functools.reduce so we can optionally add the PK filter to the
+            # list
+            qwargs = functools.reduce(
+                Q.__or__, (Q(**{f"{f}__icontains": q}) for f in fields)
+            )
+
+            qs = qs.filter(qwargs)
```
Comment (lucyb) [inline staff/views/job_requests.py:51]: Filter by identifier isn't working for me for some reason

**Review by lucyb — COMMENTED** [review]
As I understand it, Job Runner provides the Job ID, so this gets searched for a lot when debugging issues. Can this be added to the filter in the staff area too please?

File: templates/staff/job_request_detail.html [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+{% block jumbotron %}
+<div class="jumbotron jumbotron-fluid jumbotron--danger pt-md-2">
+  <div class="container">
+    <h1 class="display-4">{{ job_request.pk }}</h1>
+
+    <ul class="list-unstyled lead">
+      <li>
+        <strong>Identifier:</strong> <code>{{ job_request.identifier }}</code>
+      </li>
```
[the remark this answers is not in the document]
Comment (lucyb) [reply templates/staff/job_request_detail.html:31]: Oh sorry, that comment was supposed to be about the Job Request List page.. so we can quickly see more information about the job request without having to click into each one. The workspace name is more useful to me than the person who made the request.

File: templates/staff/job_request_list.html
Comment (lucyb) [inline templates/staff/job_request_list.html:1]: One final thing... should the filters that have been selected be visible on the left hand side of this page? At the moment, it's not obvious that I'm viewing a filtered list, except for a "clear all" option

Comment (ghickman) [reply templates/staff/job_request_list.html:1]: There are no filter controls currently.  We don't have any dropdowns-with-search that I could throw into this page.  The search queries aren't displayed on the page in any of our list views, but thinking about it that would be a nice addition, captured in #3326

**Comment by ghickman** [comment]
I've also added Job identifiers to the search now too.

**Review by lucyb — APPROVED** [review]

**Pull request merged**